src/igvfd/types/file_set.py

```diff
@@ -457,6 +466,37 @@ def functional_assay_mechanisms(self, request, input_file_sets=None):
                 mechanism_objects.extend(file_set_object.get('functional_assay_mechanisms', []))
         return list(set(mechanism_objects))
 
+    @calculated_property(
+        schema={
+            'title': 'Workflows',
+            'description': 'A workflow for computational analysis of genomic data. A workflow is made up of analysis steps.',
+            'type': 'array',
+            'notSubmittable': True,
+            'uniqueItem': True,
+            'minItems': 1,
+            'items': {
+                'title': 'Workflow',
+                'type': 'string',
+                'linkTo': 'Workflow'
+            }
+        }
+    )
+    def workflows(self, request, files=None):
+        analysis_set_workflows_set = set()
+        # Get a list of file objects
+        file_objs = get_file_objs_from_files(request=request, files=files)
+        for file_obj in file_objs:
+            if file_obj.get('analysis_step_version'):
+                # Get analysis step version and request the object
+                analysis_step_version_obj = request.embed(file_obj.get('analysis_step_version'), '@@object')
+                # Get analysis step and request the object
+                if analysis_step_version_obj.get('analysis_step'):
+                    analysis_step_obj = request.embed(analysis_step_version_obj.get('analysis_step'), '@@object')
+                    # Get workflow and add to the set
+                    if analysis_step_obj.get('workflow'):
+                        analysis_set_workflows_set.add(analysis_step_obj.get('workflow'))
```

Comment:
You can avoid redundantly getting the property by assigning it as a variable.

Like:
analysis_step_version = file_obj.get('analysis_step_version') 
if analysis_step_version:
analysis_step_version_object = request.embed(analysis_step_version, '@@object')